src/Model/CreatePaymentCustomer.php

```diff
@@ -21,35 +19,29 @@ final class CreatePaymentCustomer
     private $shippingAddress;
 
     /**
-     * @param string|null $name
-     * @param string|null $surname
+     * @note At least one of $email and $phone is required.
+     *
      * @param string|null $email
      * @param string|null $phone - customer phone in international format max 15 numeric chars https://en.wikipedia.org/wiki/MSISDN
      */
-    public function __construct($name, $surname, $email, $phone, Address $billingAddress = null, Address $shippingAddress = null)
+    public function __construct(string $name, string $surname, $email, $phone, Address $billingAddress = null, Address $shippingAddress = null)
     {
-        $this->name = $name === null ? null : new StringValue($name);
-        $this->surname = $surname === null ? null : new StringValue($surname);
-        $this->email = $email === null ? null : new StringValue($email);
-        $this->phone = $phone === null ? null : new PhoneNumber($phone);
+        $this->name = $name;
+        $this->surname = $surname;
+        $this->email = $email === null && $phone !== null ? null : new StringValue($email);
+        $this->phone = $phone === null && $email !== null ? null : new PhoneNumber($phone);
```

Comment:
I do not understand this condition, conversion into value object should be same. We have problem only if `phone` and `email` is both `null` and throwing https://www.php.net/manual/en/class.invalidargumentexception.php will be better solution.

Reply:
It allows `null` e-mail, or phone, but not both - in that case it throws that e-mail must not be `null`. Extra exception may be more readable, I updated it.